### PR TITLE
Basic controller vibration

### PR DESCRIPTION
**Pull request by devildrake**
Established basic vibration starting (with strength and duration parameters)

Made vibrations accessible through GameplaySystems

### CODE REVIEW

File: Project/Source/Modules/ModuleInput.h [hunk 1]
```diff
@@ -28,15 +29,33 @@ class PlayerController {
 		controller = SDL_GameControllerOpen(index_);
 		SDL_Joystick* j = SDL_GameControllerGetJoystick(controller);
 		joystickIndex = SDL_JoystickInstanceID(j);
+		haptic = SDL_HapticOpenFromJoystick(j);
+		if (haptic == NULL)
+			return;
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleInput.h:34]: The retursn should be on the same line.

File: Project/Source/Modules/ModuleInput.h [hunk 2]
```diff
@@ -28,15 +29,33 @@ class PlayerController {
...
 			controller = NULL;
 		}
 	}
 
+	//Strength is a normalized value, from 0 strength to full strength (1) duration in millis
+	void StartSimpleControllerVibration(float strength, float duration) {
+		if (haptic == NULL) return;
+
+		if (SDL_HapticRumbleInit(haptic) != 0)
+			return;
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleInput.h:53]: The retursn should be on the same line.

File: Project/Source/Modules/ModuleInput.h [hunk 3]
```diff
@@ -28,15 +29,33 @@ class PlayerController {
...
+
+		if (SDL_HapticRumblePlay(haptic, strength, duration) != 0)
+			return;
```
Comment (JaimeBea) [inline Project/Source/Modules/ModuleInput.h:56]: The retursn should be on the same line.

**Review by JaimeBea — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by magalenyo — APPROVED** [review]
As a suggestion, I think it would be nice to have the intensity in an Enum that is also Exposed to GameplaySystems so anyone can easily understand what the intensity range is. For example, with made up numbers. SOFT: 0.1, REGULAR: 0.5, STRONG: 1. There's no need to implement this tho, it's quite a simple implementation and you point the range out in the documentation

**Review by JaimeBea — APPROVED** [review]

**Review by magalenyo — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Pull request merged**